manta-crypto/src/eclair/alloc.rs

```diff
@@ -144,6 +224,90 @@ where
     }
 }
 
+impl<T, M, COM> Variable<M, COM> for Box<T>
+where
+    COM: ?Sized,
+    T: Variable<M, COM>,
+{
+    type Type = Box<T::Type>;
+
+    #[inline]
+    fn new_unknown(compiler: &mut COM) -> Self {
+        Self::new(compiler.allocate_unknown())
+    }
+
+    #[inline]
+    fn new_known(this: &Self::Type, compiler: &mut COM) -> Self {
+        Self::new(T::new_known(this, compiler))
+    }
+}
+
+impl<T, const N: usize, M, COM> Variable<M, COM> for [T; N]
```

Comment:
```suggestion
impl<T, M, const N: usize, COM> Variable<M, COM> for [T; N]
```

Reply:
Here `T` and `N` specify the type so they should go first.